pkg/utils/tar.go

```diff
@@ -16,6 +18,10 @@ type FileToSave struct {
 	Destination string
 }
 
+// MaxDecompressedContentSize is the maximum size of a file that can be written to disk after decompression.
+// This is to prevent a DoS attack by unpacking a compressed file that is too big to be decompressed.
+const MaxDecompressedContentSize = 1024 * 1024 * 1024 // 1 GB
```

Comment:
Is there a reason why you set it to 1GB?

Reply:
here are the sizes of image layers used in eden harness:
```
➜  docker history lfedge/adam:0.0.43         
IMAGE          CREATED         CREATED BY                                      SIZE      COMMENT
5a07fc951aec   13 months ago   ENTRYPOINT ["/bin/adam"]                        0B        buildkit.dockerfile.v0
<missing>      13 months ago   WORKDIR /adam                                   0B        buildkit.dockerfile.v0
<missing>      13 months ago   COPY /adam/swaggerui/swagger.json ./swaggeru…   1.28kB    buildkit.dockerfile.v0
<missing>      13 months ago   ADD swaggerui ./swaggerui/ # buildkit           18.5MB    buildkit.dockerfile.v0
<missing>      13 months ago   COPY /out/ / # buildkit                         21.6MB    buildkit.dockerfile.v0
➜  docker history redis:7           
IMAGE          CREATED       CREATED BY                                      SIZE      COMMENT
bdff4838c172   2 weeks ago   CMD ["redis-server"]                            0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   EXPOSE map[6379/tcp:{}]                         0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   ENTRYPOINT ["docker-entrypoint.sh"]             0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY docker-entrypoint.sh /usr/local/bin/ # …   661B      buildkit.dockerfile.v0
<missing>      2 weeks ago   WORKDIR /data                                   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   VOLUME [/data]                                  0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   RUN /bin/sh -c mkdir /data && chown redis:re…   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   RUN /bin/sh -c set -eux;   savedAptMark="$(a…   58.8MB    buildkit.dockerfile.v0
<missing>      2 weeks ago   ENV REDIS_DOWNLOAD_SHA=8d104c26a154b29fd67d6…   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   ENV REDIS_DOWNLOAD_URL=http://download.redis…   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   ENV REDIS_VERSION=7.2.4                         0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   RUN /bin/sh -c set -eux;  savedAptMark="$(ap…   4.12MB    buildkit.dockerfile.v0
<missing>      2 weeks ago   ENV GOSU_VERSION=1.17                           0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   RUN /bin/sh -c set -eux;  apt-get update;  a…   5.08kB    buildkit.dockerfile.v0
<missing>      2 weeks ago   RUN /bin/sh -c set -eux;  groupadd -r -g 999…   4.3kB     buildkit.dockerfile.v0
<missing>      2 weeks ago   /bin/sh -c #(nop)  CMD ["bash"]                 0B        
<missing>      2 weeks ago   /bin/sh -c #(nop) ADD file:9deb26e1dbc258df4…   74.8MB    
➜  docker history registry:2.7 
IMAGE          CREATED       CREATED BY                                      SIZE      COMMENT
b8604a3fe854   2 years ago   /bin/sh -c #(nop)  CMD ["/etc/docker/registr…   0B        
<missing>      2 years ago   /bin/sh -c #(nop)  ENTRYPOINT ["/entrypoint.…   0B        
<missing>      2 years ago   /bin/sh -c #(nop) COPY file:507caa54f88c1f38…   155B      
<missing>      2 years ago   /bin/sh -c #(nop)  EXPOSE 5000                  0B        
<missing>      2 years ago   /bin/sh -c #(nop)  VOLUME [/var/lib/registry]   0B        
<missing>      2 years ago   /bin/sh -c #(nop) COPY file:4544cc1555469403…   295B      
<missing>      2 years ago   /bin/sh -c #(nop) COPY file:21256ff7df5369f7…   20.1MB    
<missing>      2 years ago   /bin/sh -c set -ex     && apk add --no-cache…   549kB     
<missing>      2 years ago   /bin/sh -c #(nop)  CMD ["/bin/sh"]              0B        
<missing>      2 years ago   /bin/sh -c #(nop) ADD file:efe2d94a88cdbbd01…   5.62MB    
➜  docker history lfedge/eve:11.7.0-kvm-amd64
IMAGE          CREATED       CREATED BY                                      SIZE      COMMENT
e35135e8d483   2 weeks ago   ENTRYPOINT ["/runme.sh"]                        0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   WORKDIR /bits                                   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   LABEL org.lfedge.eci.artifact.initrd=/bits/i…   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   LABEL org.lfedge.eci.artifact.root=/bits/roo…   0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   RUN /bin/sh -c mkdir /in /out # buildkit        0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY runme.sh / # buildkit                      7.5kB     buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY installer /bits # buildkit                 272MB     buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY /out/ / # buildkit                         23.8MB    buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY / / # buildkit                             1.78MB    buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY / / # buildkit                             16.2MB    buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY / / # buildkit                             19.7MB    buildkit.dockerfile.v0
<missing>      2 weeks ago   ENTRYPOINT ["/make-config"]                     0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   WORKDIR /                                       0B        buildkit.dockerfile.v0
<missing>      2 weeks ago   COPY /out / # buildkit                          5.66MB    buildkit.dockerfile.v0
```
as we can see EVE has the biggest layer at 272MB (which is also the unpacked size) - so I figured 1GB should suffice